src/revivalpmmp/pureentities/entity/monster/walking/Wolf.php

```diff
@@ -401,12 +404,36 @@ public function isFriendly(): bool {
     }
 
     /**
+     * Sets the collar color when tamed
+     *
+     * @param $collarColor
+     */
+    public function setCollarColor($collarColor) {
+        if ($this->isTamed()) {
+            $this->namedtag->CollarColor = new ByteTag(self::NBT_KEY_COLLAR_COLOR, $collarColor); // set collar color
```

Comment:
The NBT key should be correct. Don't know about the data property. Actually I can't find a "real" data property for the collar color - therefore is used DATA_COLOUR. And i tried also undefined data keys. Just the COLOUR data key worked. As long as there's no other data key I use it as it is.

And why should i take a look the code? It's java and it's the Minecraft PC version. It's a COMPLETELY different implementation.